src/ast/mod.rs

```diff
@@ -1749,6 +1749,40 @@ pub enum FromTable {
     WithoutKeyword(Vec<TableWithJoins>),
 }
 
+/// A variable in a `SET` statement.
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum SetVariables {
```

Comment:
This syntactic pattern of `{ thing | (thing1, thing2, ...) }` is somewhat common in SQL so I was thinking we could actually make this generic like I did in my PR over here: https://github.com/sqlparser-rs/sqlparser-rs/pull/1257/files#diff-06ccb0b0ac8767ba02e52e9776bd9acfc212c87f8e943b06d093233c46b9e15aR830

An advantage of having a single type for this is that we could later add an `IntoIter` impl so consumers of these types don't need to deal with the multiple variants here.

Reply:
Nice yeah I think that makes a lot of sense!

Reply:
is the idea to do this in a follow on PR(s)?

Reply:
Yeah, I can open a PR for it after this one merges

Reply:
Ah right the other PR's merged I can replace the type while rebasing :+1:

Reply:
Rebased and replaced the type!